core/src/main/java/org/elasticsearch/action/delete/TransportDeleteAction.java

```diff
@@ -145,7 +145,8 @@ protected DeleteResponse newResponseInstance() {
 
         assert request.versionType().validateVersionForWrites(request.version());
         return new WriteResult<>(
-            new DeleteResponse(indexShard.shardId(), request.type(), request.id(), delete.version(), delete.found()),
+            new DeleteResponse(indexShard.shardId(), request.type(), request.id(), delete.version(), delete.found(),
+                               Math.max(1L, delete.endTime() - delete.startTime())),
```

Comment:
The `Math.max` calls here and throughout are unnecessary; `System#nanoTime` is relative and will not go backwards.